app/connectors/PayApiConnector.scala

```diff
@@ -46,4 +60,20 @@ class PayApiConnector @Inject()(http: WsAllMethods, configDecorator: ConfigDecor
       }
     }
   }
+
+  def findPayments(
+    utr: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[PaymentSearchResult]] = {
+    val url = s"${configDecorator.getPaymentsUrl}/$utr"
+
+    http.GET(url) flatMap { response =>
+      response.status match {
+        case OK =>
+          Future.successful(Some(response.json.as[PaymentSearchResult]))
+        case NOT_FOUND =>
```

Comment:
any status other than 2xx cant happen here due to HttpReads being the default reads.